modules/registry-configuring-registry-storage-rhodf-cephrgw.adoc

```diff
@@ -0,0 +1,104 @@
+
+:_content-type: PROCEDURE
+[id="registry-configuring-registry-storage-rhodf-cephrgw_{context}"]
+= Configuring the Image Registry Operator to use Ceph RGW storage with Red Hat OpenShift Data Foundation
+
+{rh-storage-first} integrates multiple storage types that you can use with the internal image registry:
+
+* Ceph, a shared and distributed file system and on-premises object storage
+* NooBaa, providing a Multicloud Object Gateway
+
+This procedure outlines the procedure to  configure the image registry to use Ceph RGW storage.
+
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have access to the {product-title} web console.
+* You installed the `oc` CLI.
+* Persistent storage provisioned for your cluster, such as {rh-storage}.
+
+.Procedure
+
+. Create the object bucket claim using the `ocs-storagecluster-ceph-rgw` storage class. For example:
++
+[source,terminal]
+----
+cat <<EOF | oc apply -f -
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: rgwtest
+  namespace: openshift-storage
+spec:
+  storageClassName: ocs-storagecluster-ceph-rgw
+  generateBucketName: rgwtest
+EOF
+----
+
+. Get the bucket name by entering the following command:
++
+[source,terminal]
+----
+$ bucket_name=$(oc get obc  rgwtest -o jsonpath='{.spec.bucketName}')
+----
+
+. Get the AWS credentials by entering the following commands:
++
+[source,terminal]
+----
+$ AWS_ACCESS_KEY_ID=$(oc get secret -n openshift-storage rgwtest -o yaml | grep -w "AWS_ACCESS_KEY_ID:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
++
+[source,terminal]
+----
+$ AWS_SECRET_ACCESS_KEY=$(oc get secret -n openshift-storage rgwtest -o yaml | grep -w "AWS_SECRET_ACCESS_KEY:" | head -n1 | awk '{print $2}' | base64 --decode)
+----
+
+. Create a secret with the AWS credentials for the new bucket under `openshift-image-registry project` by entering the following command:
++
+[source,terminal]
+----
+$ oc create secret generic image-registry-private-configuration-user --from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=${AWS_ACCESS_KEY_ID} --from-literal=REGISTRY_STORAGE_S3_SECRETKEY=${AWS_SECRET_ACCESS_KEY} --namespace openshift-image-registry
+----
+
+. Create a encryption route for Ceph RGW by entering the following command:
++
+[source,terminal]
+----
+$ oc create route reencrypt <route_name> --service=rook-ceph-rgw-ocs-storagecluster-cephobjectstore --port=https -n openshift-storage
+----
++
+.. Get the route host by enterting the following command:
++
+[source,terminal]
+----
+$ route_host=$(oc get route <route_name> -n openshift-storage -o=jsonpath='{.spec.host}')
+----
+. Create config map that uses an ingress certificate by entering the following commands:
++
+[source,terminal]
+----
+$ oc extract secret/router-certs-default  -n openshift-ingress  --confirm
+----
++
+[source,terminal]
+----
+$ oc create configmap user-ca-bundle --from-file=ca-bundle.crt=./tls.crt  -n openshift-config
```

Comment:
afaik `user-ca-bundle` is a special name that is used by multiple components, can we use something like `image-registry-s3-bundle`?

Reply:
Updated